contracts/gnosis_safe_modules/DelayedRecoveryModule.sol

```diff
@@ -0,0 +1,627 @@
+pragma solidity ^0.5.0;
+
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./GnosisSafeModule.sol";
+import "./GnosisSafeModuleManagerInterface.sol";
+
+/**
+ * @title Allows to replace an owner without Safe confirmations
+ *        if the recovery owner and the recovery controller approves
+ *        replacement.
+ *
+ * @dev The contract is a module for gnosis safe multisig contract.
+ *      Gnosis safe multisig's modules assume a common interface which
+ *      in the current gnosis safe implementation is inside Module.sol
+ *      contract of Gnosis Safe. Instead of inheriting from this contract
+ *      (not to include gnosis contracts into build process)
+ *      GnosisSafeModule.sol and GnosisSafeModuleManagerInterface.sol contracts
+ *      are introduced that contains the required public interfaces.
+ */
+contract DelayedRecoveryModule is GnosisSafeModule {
+
+    /* Events */
+
+    event RecoveryInitiated(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event RecoveryExecuted(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event RecoveryAborted(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event ResetRecoveryOwner(
+        address _oldRecoveryOwner,
+        address _newRecoveryOwner
+    );
+
+
+    /* Constants */
+
+    string public constant NAME = "Delayed Recovery Module";
+
+    string public constant VERSION = "0.1.0";
+
+    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = keccak256(
+        "EIP712Domain(address delayedRecoveryModule)"
+    );
+
+    bytes32 public constant INITIATE_RECOVERY_STRUCT_TYPEHASH = keccak256(
+        "InitiateRecoveryStruct(address prevOwner,address oldOwner,address newOwner)"
+    );
+
+    bytes32 public constant EXECUTE_RECOVERY_STRUCT_TYPEHASH = keccak256(
```

Comment:
`recoveryOwner` does not nee sign once the delay is passed; that way the execution can be automated by the controller and the wallet is ready when the user returns

Reply:
Agreed. Done.